File: CS-Examples/02_Data/AddListBoxControl.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;
using Spire.Xls.Core;

namespace AddListBoxControl
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load the Excel document from the specified file path
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ExcelSample_N1.xlsx");

            // Get the first worksheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set text for cells A7 to A12
            sheet.Range["A7"].Text = "Beijing";
            sheet.Range["A8"].Text = "New York";
            sheet.Range["A9"].Text = "ChengDu";
            sheet.Range["A10"].Text = "Paris";
            sheet.Range["A11"].Text = "Boston";
            sheet.Range["A12"].Text = "London";

            // Set text and formatting for cell C13
            sheet.Range["C13"].Text = "City :";
            sheet.Range["C13"].Style.Font.IsBold = true;

            // Add a listbox control to the worksheet
            IListBox listBox = sheet.ListBoxes.AddListBox(13, 4, 100, 80);
            // Set the selection type to single (allows only one item to be selected)
            listBox.SelectionType = SelectionType.Single;
            // Set the initially selected index in the listbox
            listBox.SelectedIndex = 2;
            // Enable 3D shading for the listbox
            listBox.Display3DShading = true;
            // Specify the range to populate the listbox with data
            listBox.ListFillRange = sheet.Range["A7:A12"];

            //Specify the filename for the resulting Excel file
            string output = "InsertListBoxControl_out.xlsx";

            // Save the workbook to the specified file in Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}

File: CS-Examples/02_Data/AddListBoxControl.vb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports Spire.Xls.Core

Namespace AddListBoxControl
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Excel workbook object.
            Dim workbook As New Workbook()

            ' Load an existing Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ExcelSample_N1.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Set the value of cell A7 to "Beijing"
            sheet.Range("A7").Text = "Beijing"
            ' Set the value of cell A8 to "New York".
            sheet.Range("A8").Text = "New York"
            ' Set the value of cell A9 to "ChengDu".
            sheet.Range("A9").Text = "ChengDu"
            ' Set the value of cell A10 to "Paris".
            sheet.Range("A10").Text = "Paris"
            ' Set the value of cell A11 to "Boston".
            sheet.Range("A11").Text = "Boston"
            ' Set the value of cell A12 to "London".
            sheet.Range("A12").Text = "London"
            ' Set the value of cell C13 to "City :".
            sheet.Range("C13").Text = "City :"
            ' Apply bold font style to the text in cell C13.
            sheet.Range("C13").Style.Font.IsBold = True

            ' Add a listbox control at position (13, 4) with width 100 and height 80.
            Dim listBox As IListBox = sheet.ListBoxes.AddListBox(13, 4, 100, 80)
            ' Set the selection type of the listbox to single.
            listBox.SelectionType = SelectionType.Single
            ' Set the selected index of the listbox to 2 (selects the third item).
            listBox.SelectedIndex = 2
            ' Enable 3D shading for the listbox.
            listBox.Display3DShading = True
            ' Set the data range for the listbox to cells A7 to A12.
            listBox.ListFillRange = sheet.Range("A7:A12")

            ' Specify the output file name.
            Dim output As String = "InsertListBoxControl_out.xlsx"
            ' Save the modified workbook to the specified file with Excel 2013 format.
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
